docs/docs/how-to/dbauth-passwordless.md

```diff
@@ -0,0 +1,644 @@
+  loginToken          String
+  loginTokenExpiresAt DateTime?
+  salt                String?
+}
+```
+
+Make note of the optional `salt` field.
+
+Once you've made the changes, you'll need to migrate your database.
+
+```bash
+yarn rw prisma migrate dev
+```
+
+### 2. Setting up the generateToken function
+Next, we need to create a function that will generate a token and an expiration date.
+
+If you followed the tutorial, you might not have a `/api/src/services/users/users.js` file.  If that's the case, you can create it.
+
+```bash
+yarn rw g service users
+```
+
+Now that you have the file, let's add the `generateToken` function.
+
+```javascript {21} title="/api/src/services/users/users.js"
+// add this to the bottom of the file
+export const generateLoginToken = async ({ email }) => {
+  try {
+    // look up if the user exists
+    let lookupUser = await db.user.findFirst({ where: { email } })
+    if (!lookupUser) return { message: 'Login Request received' }
+    // here we're going to generate a random password of 6 numbers
+    let randomNumber = (() => {
+      let random = CryptoJS.lib.WordArray.random(6)
+      let randomString = random.toString()
+      let sixDigitNumber = randomString.replace(/\D/g, '')
+      if (sixDigitNumber.length < 6) {
+        sixDigitNumber = sixDigitNumber.padStart(6, '0')
+      }
+      if (sixDigitNumber.length > 6) {
+        sixDigitNumber = sixDigitNumber.slice(0, 6)
+      }
+      return sixDigitNumber.toString()
+    })()
+    console.log({ randomNumber }) // email the user this number
+    let salt = CryptoJS.lib.WordArray.random(30)
+    let loginToken = CryptoJS.PBKDF2(randomNumber, salt, {
+      keySize: 256 / 32,
+    }).toString()
+    // now we'll update the user with the new salt and loginToken
+    let loginTokenExpiresAt = new Date()
+    loginTokenExpiresAt.setMinutes(loginTokenExpiresAt.getMinutes() + 15)
+    let data = {
+        salt,
+        loginToken,
+        loginTokenExpiresAt,
+    }
+    await db.user.update({
+      where: { id: lookupUser.id },
+      data
+    })
+
+    return { message: 'Login Request received' }
+  } catch (error) {
+    console.log({ error })
+    throw new UserInputError(error.message)
+  }
+}
+
+```
+### 3. Add generateToken to the SDL and secure loginToken
+
+In addition to the new function, we need to add it to the sdl file.  While we're here let's also ensure we do not expose the loginToken.
+
+```javascript {21} title="/api/src/graphql/users.sdl.js"
+type User {
+  id: Int!
+  name: String
+  email: String!
+}
+input CreateUserInput {
+  name: String
+  email: String!
+}
+input UpdateUserInput {
+  name: String
+  email: String!
+}
+type userTokenResponse {
+  message: String!
+}
+type Mutation {
+  createUser(input: CreateUserInput!): User! @requireAuth
+  updateUser(id: Int!, input: UpdateUserInput!): User! @requireAuth
+  deleteUser(id: Int!): User! @requireAuth
+  generateToken(email: String!): userTokenResponse! @skipAuth
+}
+```
```

Comment:
I broke out the SDL stuff into it's own section explicitly listing the fields and noting to exclude `loginToken`